lib/dynamoid/adapter.rb

```diff
@@ -19,7 +19,7 @@ def tables
       if !@tables_.value
         @tables_.swap{|value, args| benchmark('Cache Tables') {list_tables}}
       end
-      @tables_.value
+      @tables_.value || []
```

Comment:
Hey everyone! I'm also looking to use Dynamoid with Rails 5 and so I fork it and apply all this changes and started to build a test to cover this situation (FYI I did not experience this issue though). But looking into the code I got confuse with this change, because the
```ruby
if !@tables_.value
  @tables_.swap{|value, args| benchmark('Cache Tables') {list_tables}}
end
```
should do exactly the same as this `@tables_.value || []` because the output of the method `list_tables` of the `Dynamoid::AdapterPlugin::AwsSdkV2` is an empty array. So I imagine that's this is depending at the end on the `Aws::DynamoDB::Client` version that you are using, because the `Dynamoid::AdapterPlugin::AwsSdkV2` send that call to the `client` that it's a `Aws::DynamoDB::Client`.

Bottom line the response of the `Aws::DynamoDB::Client.list_tables` is a `Aws::DynamoDB::Types::ListTablesOutput` struct with a `table_names` array.

I'm using `aws-sdk-core (2.2.33)` and `aws-sdk-resources (2.2.33)`.
Any other thought?